src/core/thread/mle.cpp

```diff
@@ -1497,8 +1510,23 @@ ThreadError Mle::HandleDataResponse(const Message &aMessage, const Ip6::MessageI
     SuccessOrExit(error = Tlv::GetTlv(aMessage, Tlv::kLeaderData, sizeof(leaderData), leaderData));
     VerifyOrExit(leaderData.IsValid(), error = kThreadError_Parse);
 
-    diff = leaderData.GetDataVersion() - mNetworkData.GetVersion();
-    VerifyOrExit(diff > 0, ;);
+    if (mRetrieveNewNetworkData)
+    {
+        mRetrieveNewNetworkData = false;
+    }
+    else
+    {
+        diff = leaderData.GetDataVersion() - mNetworkData.GetVersion();
+
+        if (((mDeviceMode & ModeTlv::kModeRxOnWhenIdle) == 0) && (diff != 0))
+        {
+            SetLeaderData(leaderData.GetPartitionId(), leaderData.GetWeighting(), leaderData.GetLeaderRouterId());
```

Comment:
hi @jwhui , SetLeaderData for rx-on-when-idle child is done when HandleAdvertisement(). And what it still need is to retrieve new networkdata. so you think it is better to put SetLeaderData for rx-on-when-idle child  here?